Update the version to 2.1.0 and rename the provisioner name (#12)
projectsyn/component-csi-cloudscale: docs/modules/ROOT/pages/how-tos/upgrade-1.x-to-2.x.adoc

= Upgrade from 1.x to 2.x

This guide describes the steps to perform an upgrade of the component from version 1.x to 2.x.

====
Requirements

* `commodore`
* `kubectl`
* `shell`
====

== Step-by-step guide

The essential part of the upgrade is the renaming of the provisioner name from `ch.cloudscale.csi` to `csi.cloudscale.ch`.

> This implies that all `PersistentVolumes` require an changed annotation and all `StorageClasses` need to be recreated with the new provisioner name.

Reviewer: ```suggestion
This implies that all `PersistentVolumes` require a changed annotation and all `StorageClasses` need to be recreated with the new provisioner name.
```

> The upgrade itself does not affect the data path, so no interrupt of IO should be there for applications.

Reviewer: ```suggestion
The upgrade itself does not affect the data path, so applications should not experience IO interrupts during the upgrade.
```

> However, during the upgrade volume changes might not work, so doing that off hours is suggested.

Reviewer: ```suggestion
However, during the upgrade, volume provisioning and volume changes might not work for short periods of time, so it's strongly recommended to perform the upgrade during off hours.
```

> All steps should be done together.

Reviewer: ```suggestion
To minimize the visible impact of the upgrade, all steps should be done together.
```
